code/datums/elements/mechanical_repair.dm

```diff
@@ -43,27 +43,27 @@
 
 	// Handles welder repairs on human limbs
 	if(I.tool_behaviour == TOOL_WELDER)
-		do
-			if(I.use_tool(target, user, 0, volume=50, amount=1))
-				if(user == target)
-					user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [target == user ? "[p_their()]" : "[target]'s"] [parse_zone(affecting.body_zone)].</span>",
-					"<span class='notice'>You start fixing some of the dents on [target == user ? "your" : "[target]'s"] [parse_zone(affecting.body_zone)].</span>")
-					if(!do_after(user, 1.5 SECONDS, target))
-						return COMPONENT_NO_AFTERATTACK
-		while (item_heal_robotic(target, user, 15, 0, affecting) && user.is_zone_selected(selected_zone) && !QDELETED(I))
+		if(I.use_tool(target, user, 0, volume=50, amount=0))
+		//Just to check if the tool is even on. The strange order here requires strange solutions
+			do
+				user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [target == user ? "[p_their()]" : "[target]'s"] [parse_zone(affecting.body_zone)].</span>",
+				"<span class='notice'>You start fixing some of the dents on [target == user ? "your" : "[target]'s"] [parse_zone(affecting.body_zone)].</span>")
+				if(!do_after(user, 1.5 SECONDS, target))
+					return COMPONENT_NO_AFTERATTACK
+			while ((I.use_tool(target, user, 0, volume=50, amount=1) ? item_heal_robotic(target, user, 15, 0, affecting) : FALSE) && user.is_zone_selected(selected_zone) && !QDELETED(I))
```

Comment:
```suggestion
			if(!I.use_tool(target, user, 0, volume=50, amount=1)
				break
			while (item_heal_robotic(target, user, 15, 0, affecting && user.is_zone_selected(selected_zone) && !QDELETED(I))
```
This makes for a little cleaner code by doing the same thing; but less on the same line.